File: 2580-count-ways-to-group-overlapping-ranges/2580-count-ways-to-group-overlapping-ranges.py
class Solution:
    def countWays(self, A: List[List[int]]) -> int:
        pre = -1
        res = 0
        for a,b in sorted(A):
            res += pre < a
            pre = max(pre, b)
        return pow(2, res, 10**9+7)
        

File: 2580-count-ways-to-group-overlapping-ranges/2580-count-ways-to-group-overlapping-ranges.cpp
class Solution {
public:
    int countWays(vector<vector<int>>& ranges) {
        int res = 1, last = -1, mod = 1e9 + 7;
        sort(ranges.begin(), ranges.end());
        for (auto r: ranges) {   
            if (last < r[0])
                res = res * 2 % mod;
            last = max(last, r[1]);
        }
        return res;
    }
};